src/Controls/tests/TestCases.HostApp/Issues/Issue29421.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections.ObjectModel;
+
+namespace Maui.Controls.Sample.Issues;
+
+[Issue(IssueTracker.Github, 29421, "KeepScrollOffset Not Working as Expected in CarouselView", PlatformAffected.UWP)]
+public class Issue29421 : ContentPage
+{
+	public Issue29421()
+	{
+		var verticalStackLayout = new VerticalStackLayout();
+		var carouselItems = new ObservableCollection<string>
+		{
+			"Item 0",
+			"Item 1",
+			"Item 2",
+			"Item 3",
+			"Item 4",
+		};
+
+		CarouselView carouselView = new CarouselView
+		{
+			ItemsSource = carouselItems,
+			AutomationId = "carouselview",
+			ItemsUpdatingScrollMode = ItemsUpdatingScrollMode.KeepScrollOffset,
+			HeightRequest = 300,
+			ItemTemplate = new DataTemplate(() =>
+			{
+				var grid = new Grid
+				{
+					Padding = 10
+				};
+
+				var label = new Label
+				{
+					VerticalOptions = LayoutOptions.Center,
+					HorizontalOptions = LayoutOptions.Center,
+					FontSize = 18,
+				};
+				label.SetBinding(Label.TextProperty, ".");
+				label.SetBinding(Label.AutomationIdProperty, ".");
+
+				grid.Children.Add(label);
```

Comment:
Can you set a random background color to the items to easily view the item size?